drivers/drv8825/src/drv8825.rs

```diff
@@ -0,0 +1,369 @@
+    /// Enables support for step mode control and sets the initial step mode
+    ///
+    /// Consumes this instance of `DRV8825` and returns another instance that
+    /// has support for controlling the step mode. Requires the additional pins
+    /// for doing so, namely RESET, MODE0, MODE1, and MODE2. It expects the
+    /// types that represent those pins to implement [`OutputPin`].
+    ///
+    /// This method is only available when those pins have not been provided
+    /// yet. After this method has been called once, you can use
+    /// [`DRV8825::set_step_mode`] to change the step mode again.
+    pub fn enable_mode_control<
+        Reset,
+        Mode0,
+        Mode1,
+        Mode2,
+        Clk,
+        OutputPinError,
+    >(
+        self,
+        reset: Reset,
+        mode0: Mode0,
+        mode1: Mode1,
+        mode2: Mode2,
+        step_mode: StepMode,
+        clock: &Clk,
+    ) -> Result<
+        DRV8825<(), (), (), Reset, Mode0, Mode1, Mode2, Step, Dir>,
+        ModeError<OutputPinError>,
+    >
+    where
+        Reset: OutputPin<Error = OutputPinError>,
+        Mode0: OutputPin<Error = OutputPinError>,
+        Mode1: OutputPin<Error = OutputPinError>,
+        Mode2: OutputPin<Error = OutputPinError>,
+        Clk: Clock,
+    {
+        let mut self_ = DRV8825 {
+            enable: self.enable,
+            fault: self.fault,
+            sleep: self.sleep,
+            reset,
+            mode0,
+            mode1,
+            mode2,
+            step: self.step,
+            dir: self.dir,
+        };
+
+        self_.set_step_mode(step_mode, clock)?;
+
+        Ok(self_)
+    }
+}
+
+impl<Reset, Mode0, Mode1, Mode2, Step, Dir, OutputPinError> SetStepMode
+    for DRV8825<(), (), (), Reset, Mode0, Mode1, Mode2, Step, Dir>
+where
+    Reset: OutputPin<Error = OutputPinError>,
+    Mode0: OutputPin<Error = OutputPinError>,
+    Mode1: OutputPin<Error = OutputPinError>,
+    Mode2: OutputPin<Error = OutputPinError>,
+{
+    type Error = ModeError<OutputPinError>;
+
+    /// Sets the step mode
+    ///
+    /// This method is only available, if all the pins required for setting the
+    /// step mode have been provided using [`DRV8825::enable_mode_control`].
+    fn set_step_mode<Clk: Clock>(
+        &mut self,
+        step_mode: StepMode,
+        clock: &Clk,
+    ) -> Result<(), Self::Error> {
+        // 7.6 Timing Requirements (page 7)
+        // https://www.ti.com/lit/ds/symlink/drv8825.pdf
+        const SETUP_TIME: Nanoseconds = Nanoseconds(650);
+
+        // Reset the device's internal logic and disable the h-bridge drivers.
+        self.reset
+            .try_set_low()
+            .map_err(|err| ModeError::OutputPin(err))?;
+
+        // Set mode signals. All this repetition is messy. I decided not to do
+        // anything about it and wait for the next embedded-hal alpha version,
+        // which has features that would help here.
```

Comment:
Looks like this comment was copied from `stspin220`. It's outdated, as the next embedded-hal version it was referring to has been released, and improvements have been made (before, there was no `try_set_state`, so pattern matching and `try_set_high`/`try_set_low` were required).

I suggest removing everything after "Set mode signals". I've made a note to change this over in the `stspin220` module.